Mathlib/Geometry/Manifold/Bordisms.lean

```diff
@@ -0,0 +1,202 @@
+  [isManifold: IsManifold I k M]
+  [compactSpace: CompactSpace M]
+  [boundaryless: BoundarylessManifold I M]
+  /-- `M` is `n`-dimensional, as its model space `E` is -/
+  dimension : finrank ℝ E = n
+  /-- The underlying map `M → X` of a singular `n`-manifold `(M, f)` on `X` -/
+  f : M → X
+  hf : Continuous f
+
+namespace SingularNManifold
+
+variable {X Y Z : Type*} [TopologicalSpace X] [TopologicalSpace Y] [TopologicalSpace Z]
+  {n : ℕ} {k : ℕ∞}
+  {E H M : Type*} [NormedAddCommGroup E] [NormedSpace ℝ E] [FiniteDimensional ℝ E]
+  [TopologicalSpace H] {I : ModelWithCorners ℝ E H} [TopologicalSpace M] [ChartedSpace H M]
+  [IsManifold I k M] [CompactSpace M] [BoundarylessManifold I M]
+
+instance {n : ℕ} {k : ℕ∞} {s : SingularNManifold X n k I} : TopologicalSpace s.M := s.topSpaceM
+
+instance {n : ℕ} {k : ℕ∞} {s : SingularNManifold X n k I} : ChartedSpace H s.M := s.chartedSpace
+
+instance {n : ℕ} {k : ℕ∞} {s : SingularNManifold X n k I} : IsManifold I k s.M := s.isManifold
+
+instance {n : ℕ} {k : ℕ∞} {s : SingularNManifold X n k I} : CompactSpace s.M := s.compactSpace
+
+instance {n : ℕ} {k : ℕ∞} {s : SingularNManifold X n k I} : BoundarylessManifold I s.M :=
+  s.boundaryless
+
+/-- A map of topological spaces induces a corresponding map of singular n-manifolds. -/
+-- This is part of proving functoriality of the bordism groups.
+noncomputable def map (s : SingularNManifold X n k I)
+    {φ : X → Y} (hφ : Continuous φ) : SingularNManifold Y n k I where
+  f := φ ∘ s.f
+  hf := hφ.comp s.hf
+  dimension := s.dimension
+
+@[simp]
+lemma map_f (s : SingularNManifold X n k I) {φ : X → Y} (hφ : Continuous φ) :
+    (s.map hφ).f = φ ∘ s.f :=
+  rfl
+
+lemma map_comp (s : SingularNManifold X n k I)
+    {φ : X → Y} {ψ : Y → Z} (hφ : Continuous φ) (hψ : Continuous ψ) :
+    ((s.map hφ).map hψ).f = (ψ ∘ φ) ∘ s.f := by
+  simp [Function.comp_def]
+  rfl
+
+-- Let M' and W be real C^k manifolds.
+variable {E' E'' E''' H' H'' H''' : Type*}
+  [NormedAddCommGroup E'] [NormedSpace ℝ E'] [NormedAddCommGroup E'']  [NormedSpace ℝ E'']
+  [NormedAddCommGroup E'''] [NormedSpace ℝ E''']
+  [TopologicalSpace H'] [TopologicalSpace H''] [TopologicalSpace H''']
+
+variable {M' : Type*} [TopologicalSpace M'] [ChartedSpace H' M']
+  {I' : ModelWithCorners ℝ E' H'} [IsManifold I' k M']
+  [BoundarylessManifold I' M'] [CompactSpace M'] [FiniteDimensional ℝ E']
+
+variable (M I) in
+/-- If `M` is `n`-dimensional and closed, it is a singular `n`-manifold over itself.-/
+noncomputable def refl (hdim : finrank ℝ E = n) :
+    SingularNManifold M n k I where
+  dimension := hdim
+  f := id
+  hf := continuous_id
+
+/-- If `(N, f)` is a singular `n`-manifold on `X` and `M` another `n`-dimensional manifold,
+a smooth map `φ : M → N` induces a singular `n`-manifold structure `(M, f ∘ φ)` on `X`. -/
+noncomputable def comap [h : Fact (finrank ℝ E' = n)]
+    (s : SingularNManifold X n k I)
+    {φ : M' → s.M} (hφ : ContMDiff I' I n φ) : SingularNManifold X n k I' where
```

Comment:
There's a funny confusion here: you are assuming that phi is `C^n`, where `n`  is the dimension. I think you meant `C^k`. But you should just assume continuous, because it's what you need.

Reply:
You're right, twice!